app.py

```diff
@@ -44,6 +46,13 @@ def init_app(self):
         # replace all non alphanumeric characters by '_'
         command_name = re.sub(r"[^0-9a-zA-Z]+", "_", command_name)
 
+        self.modal = self.get_setting("modal")
+
+        pre_publish_hook_path = self.get_setting(self.CONFIG_PRE_PUBLISH_HOOK_PATH)
+        self.pre_publish_hook = self.create_hook_instance(
+            pre_publish_hook_path, base_class=self.base_hooks.PrePublishHook
```

Comment:
You can leave out the `base_class` class here and let it inherit the generic `Hook` class since you're not adding any default implementations.